### PR TITLE
taito_l : Cleanups

### PR DESCRIPTION
**Pull request by cam900**
taito_l : Convert vram bankswitching functions to address_map_bank_device
Correct palette format is xRGBRRRRGGGGBBBB_bit0(Games using 15-bit palettes), but this format is different at current emulations.

### CODE REVIEW

File: src/mame/video/taito_l.cpp [hunk 1]
```diff
@@ -139,73 +131,30 @@ READ8_MEMBER(taitol_state::taitol_control_r)
...
-	m_gfxdecode->gfx(2)->mark_dirty(offset / 32 + 512);
-}
-
-void taitol_state::taitol_chardef1d_m( int offset )
-{
-	m_gfxdecode->gfx(2)->mark_dirty(offset / 32 + 640);
-}
-
-void taitol_state::taitol_chardef1e_m( int offset )
+WRITE8_MEMBER(taitol_state::vram_w)
```
Comment (Osso13) [inline src/mame/video/taito_l.cpp]: you forgot to declare  vram_w

File: src/mame/includes/taito_l.h [hunk 2]
```diff
@@ -28,30 +30,23 @@ class taitol_state : public driver_device
 	u8 *       m_shared_ram;
 
 	/* video-related */
-	tilemap_t *m_bg18_tilemap;
-	tilemap_t *m_bg19_tilemap;
-	tilemap_t *m_ch1a_tilemap;
-	u8 m_buff_spriteram[SPRITERAM_SIZE];
+	tilemap_t *m_bg_tilemap[0];
+	tilemap_t *m_bg_tilemap[1];
```
Comment (Osso13) [inline src/mame/includes/taito_l.h]: you're redeclaring stuff m_bg_tilemap[0] here, that you have declared the line above

**Comment by Tafoid** [comment]
All sets in taito_l.cpp
Incorrect Palettes

Examples: cubypop, evilston, kurikint, lagirl, puzznic

cubybop-diff
evilston-diff
kurikint-diff
lagirl-diff
puzznic-diff

**Comment by galibert** [comment]
Fixed cd1c29a3d16290c74f7afb0ec86ee6d144616116

**Comment by galibert** [comment]
Also, puzznic very very clearly uses 12-bits palettes, so I'll need more information before I accept a 15-bits one (without r and b inverted this time).

**Pull request merged**